### PR TITLE
Fixed: Issue #72 - The script will not crash when the player is not s…

### PR DESCRIPTION
**Pull request by dvdvideo1234**
…uperadmin

The script assumes that the entity `wire_dupeport` will be always created globally correctly.

### CODE REVIEW

**Comment by thegrb93** [comment]
I'll look at this in a bit.

File: lua/weapons/gmod_tool/stools/wire_dupeport.lua [hunk 1]
```diff
@@ -94,14 +98,13 @@ if (SERVER) then
...
+	if ( not ent:IsValid() ) then return end
 
-	local tr 	= util.GetPlayerTrace( player, player:GetAimVector() )
-	local trace 	= util.TraceLine( tr )
-	if (!trace.Hit) then return end
+	local trace = player:GetEyeTrace()
+	if ( not trace.Hit ) then return end
 
-	if (trace.Entity && trace.Entity:GetClass() == "gmod_wire_dupeport" || trace.Entity:IsPlayer()) then
+	if (trace.Entity and trace.Entity:GetClass() == "gmod_wire_dupeport" or trace.Entity:IsPlayer() ) then
```
Comment (thegrb93) [inline lua/weapons/gmod_tool/stools/wire_dupeport.lua]: I think this should be
`if (trace.Entity and (trace.Entity:GetClass() == "gmod_wire_dupeport" or trace.Entity:IsPlayer()) ) then`

Comment (dvdvideo1234) [reply lua/weapons/gmod_tool/stools/wire_dupeport.lua]: Done, thanks @thegrb93  !

Comment (dvdvideo1234) [reply lua/weapons/gmod_tool/stools/wire_dupeport.lua]: I added also ent:IsValid() to avoid getting the class of invalid ents.

**Comment by thegrb93** [comment]
Changes look fine. I'll assume its tested; will merge in a bit if not further comments.

**Comment by dvdvideo1234** [comment]
@thegrb93 I did test it [only syntactically][1]. It seems to be ok.

1: https://github.com/dvdvideo1234/ZeroBraineProjects/blob/master/GmodToolTest/main.lua

**Comment by dvdvideo1234** [comment]
I've added localization of the class and limit name to make it faster.

**Comment by thegrb93** [comment]
Can you revert that? The trade-off to messier code isn't worth it.

**Comment by dvdvideo1234** [comment]
> 
> 
> Can you revert the new local variables? The concatenations are messier (and slower).

But you do it only once at the beginning of the file, store the value in a local and use the local variables when you need to check `A:GetClass() == gsClass` instead of `A:GetClass() == "some_class"`.

**Comment by thegrb93** [comment]
That makes code harder to read and it's not making anything faster.

**Comment by dvdvideo1234** [comment]
Hello, @thegrb93

Forche-pushed to remove the global configurations

**Comment by thegrb93** [comment]
Can you change `local gsModel` to TOOL.DupePortModel

**Comment by thegrb93** [comment]
Meh, that's probably fine.

**Pull request merged**